crates/net/eth-wire/src/types/message.rs

```diff
@@ -286,7 +286,11 @@ pub trait EncodableExt {
 impl<T: Encodable> EncodableExt for Vec<T> {
     fn encode_max(&self, limit: usize) -> alloy_rlp::Result<Vec<u8>, alloy_rlp::Error> {
         let mut buffer = Vec::new();
+
         for item in self {
+            let mut header = Header { list: true, payload_length: 0 };
+            header.payload_length += item.length();
+            header.encode(&mut buffer);
```

Comment:
the item will probably also have a header, a header doesn't need to be added before each element

Reply:
I have made changes to above code. Can u check . then proceed for tests

Reply:
it looks great!